### ISSUE TITLE
[Skype For Business] Error 500 when sending proactive message to Skype for Business

### ISSUE THREAD

**Issue by gazambuja** [Microsoft/BotBuilder#4450]
## Bot Info
* SDK Platform: NodeJS
* SDK Version: ^3.13.1
* Active Channels: Skype and Skype for Business

## Issue Description
When try to send a proactive message to user using Skype for B. they return error 500 and never deliver the message. This work fine if the saved address is from Skype and not Skype for Bussines.

## Code Example
```var Promise = require('bluebird');
var builder = require('botbuilder');
const config = require('../config.json'); // Get config file.

var connector = new builder.ChatConnector({
    appId: config.env.MICROSOFT_APP_ID,
    appPassword: config.env.MICROSOFT_APP_PASSWORD
});

var bot = new builder.UniversalBot(connector);

sendProactiveMessage: function (address, txt) {
    return new Promise(function (resolve) {
        console.log("sending msg");
        
        var msg = new builder.Message().address(address);
        msg.text(txt);
        msg.textLocale('en-US');
        bot.send(msg);
        resolve(true);
    });
}

const address = mySavedAddress();
sendProactiveMessage(address, "Message Test");
```

## Expected Behavior
Deliver a message to user saved into address var.

## Actual Results
[500] Internal Server Error

```
Error: POST to 'https://webpooldm20r04.infra.lync.com/platformservice/tgt-fe86af5167ec51de8ae5c63*****137a2/botframework/v3/conversations/OTA4ZjZkZTEjc2lwOnNjYXJsZXR0QG92ZXJhY3*****UAY29t/activities/6' failed: [500] Internal Server Error
    at Request._callback (/Users/gustavoa/Documents/dev/Bot/node_modules/botbuilder/lib/bots/ChatConnector.js:559:46)
    at Request.self.callback (/Users/gustavoa/Documents/dev/Bot/node_modules/request/request.js:186:22)
    at emitTwo (events.js:106:13)
    at Request.emit (events.js:191:7)
    at Request.<anonymous> (/Users/gustavoa/Documents/dev/Bot/node_modules/request/request.js:1163:10)
    at emitOne (events.js:96:13)
    at Request.emit (events.js:188:7)
    at IncomingMessage.<anonymous> (/Users/gustavoa/Documents/dev/Bot/node_modules/request/request.js:1085:12)
    at IncomingMessage.g (events.js:292:16)
    at emitNone (events.js:91:20)
    at IncomingMessage.emit (events.js:185:7)
    at endReadableNT (_stream_readable.js:974:12)
    at _combinedTickCallback (internal/process/next_tick.js:80:11)
    at process._tickCallback (internal/process/next_tick.js:104:9)
```

**Comment by gazambuja** [comment]
If I remove the conversation object of address, and try to send a message, in my S4B I recibe a "Meet now" message. The bot create a new conversation. But how I can get the conversation ID to send a new message in this conversation?

**Comment by nodeomega** [comment]
@gazambuja You mean the conversation ID of the new "Meet Now" message, correct?

**Comment by gazambuja** [comment]
@nodeomega yea, exactly.

**Comment by cristianplesea** [comment]
Hi, although I'm coding bots in C#, maybe my discoveries will help. When receiving the 500 error on sending message to the previous (known) conversation, I explicitly create a new conversation, setting a topic name for it (the topic is what appears instead of the "Meet now" notification) and use this conversation's ID to re-send the message. It usually works, the only slight issues are with the iOS S4B client with doesn't always receive the notifications, only after opening the app (still looking for a solution to that). Here is the code I use, probably there is an equivalent which can be used in NodeJS.


            try
            {
                connector.Conversations.SendToConversation((Activity)messageActivity);
            }
            catch (Exception ex)
            {
                try
                {
                    if (messageActivity.ChannelId == "skypeforbusiness")
                    {
                        MicrosoftAppCredentials.TrustServiceUrl("https://api.skypeforbusiness.com/platformservice/botframework");

                        connector = new ConnectorClient(
                            new Uri("https://api.skypeforbusiness.com/platformservice/botframework/")); //initial entry point to talk to the Skype for Business channel
                        List<ChannelAccount> participants = new List<ChannelAccount>
                        {
                            new ChannelAccount(messageActivity.Recipient.Id, messageActivity.Recipient.Name)
                        };
                        ConversationParameters cpMessage = new ConversationParameters(true, new ChannelAccount(messageActivity.From.Id, messageActivity.From.Name), participants, messageActivity.Text /*topicName*/);
                        ConversationResourceResponse response = await connector.Conversations.CreateConversationAsync(cpMessage);

                        Logger.Info($"({correlationId}) Created new conversation for SkypeForBusiness user {messageActivity.Recipient.Id}: {response.Id}");

                        messageActivity.Conversation.Id = response.Id;
                        messageActivity.ServiceUrl = response.ServiceUrl;
                        await connector.Conversations.SendToConversationAsync((Activity)messageActivity);

                        Logger.Info($"({correlationId}) Sent message in new conversation for SkypeForBusiness user {messageActivity.Recipient.Id}: {response.Id}");
                    }
                }
                catch (Exception ex2)
                {
                    Logger.Error($"({correlationId}) EXCEPTION creating/sending new conversation in channel {messageActivity.ChannelId} for {messageActivity.Recipient.Id}: ", ex2);
                }
            }

**Comment by JasonSowers** [comment]
Thank you for opening an issue against the Bot Framework SDK v3. As part of the Bot Framework v4 release, we’ve moved all v3 work to a new repo located at https://github.com/microsoft/botbuilder-v3. We will continue to support and offer maintenance updates to v3 via this new repo.
 
From now on, https://github.com/microsoft/botbuilder repo will be used as hub, with pointers to all the different SDK languages, tools and samples repos.
 
As part of this restructuring, we are closing all tickets in this repo.
 
For defects or feature requests, please create a new issue in the new Bot Framework v3 repo found here:
https://github.com/microsoft/botbuilder-v3/issues

For Azure Bot Service Channel specific defects or feature requests (e.g. Facebook, Twilio, Teams, Slack, etc.), please create a new issue in the new Bot Framework Channel repo found here:
https://github.com/microsoft/botframework-services/issues
 
For product behavior, how-to, or general understanding questions, please use Stackoverflow.
https://stackoverflow.com/search?q=bot+framework


Thank you.
 
The Bot Framework Team
